### ISSUE TITLE
Retina Display

### ISSUE THREAD

**Issue by sandersantema** [KDE-mac/homebrew-kde#315]
# Please specify exact formula that has issue

All of the applications

# What happened (include command output)

Applications have a resolution which isn't correct for retina screens.  I tried including screenshots but for some reason (probably compression) the difference wasn't apparent. 

# Fix

A simple fix is found here: https://blog.qt.io/blog/2013/04/25/retina-display-support-for-mac-os-ios-and-x11/

I would do a PR but I don't know what would need to be changed, I myself just simply edited the plist file like mentioned in the article but QMake should be able to do it automatically. Otherwise you might want to use a script which edits the plist file.

**Comment by sl1pkn07** [comment]
or talk with upstream

**Comment by sandersantema** [comment]
I don't think this is an upstream issue (but I might very well be wrong) at least when compiling the applications yourself since I suspect it's a configurable option for qmake during the compilation. Might it be an idea to add the compilation flag to qmake in this repo? (If it exists, like I said I'm not knowledgeable at all with the build proces etc. of these apps) 

Otherwise I'll try to file a bug report at upstream (KDE as far as I know).

**Comment by sl1pkn07** [comment]
All kde projects is cmake based, no qmake.

**Comment by bmihaila** [comment]
@sandersantema you can add those config settings using `PlistBuddy` to any app. The homebrew formulas use that for changing setting after install. I did so for Dolphin which has bad font rendering in current install, with:
```
$❯ sudo /usr/libexec/PlistBuddy -c 'Add :NSHighResolutionCapable string "True"' ~/Applications/KDE/dolphin.app/Contents/Info.plist
$❯ sudo /usr/libexec/PlistBuddy -c 'Add :NSPrincipalClass string "NSApplication"' ~/Applications/KDE/dolphin.app/Contents/Info.plist
```
Unfortunately restarting the app did not improve the font rendering :/ - so I'm not sure that is enough.

One application which has a working HiDPI font rendering is Kate. It seems to have above 2 settings in its `Info.plist` but not sure if there is more than that because for dolphin it was not enough to set those.

One would need to take a look at the buildfiles for those apps and see what the difference is. The homebrew formulas at least do not apply any settings for them.

**Comment by bmihaila** [comment]
A small addition, Kate seems to set those perms during the build, see https://github.com/KDE/kate/blob/master/kate/data/MacOSXBundleInfo.plist.in#L5

**Comment by bmihaila** [comment]
@sandersantema  ok actually after restarting the machine the apps that got the treatment as described in https://github.com/KDE-mac/homebrew-kde/issues/315#issuecomment-578790011 look good now! \o/
It seems I underestimated the caching of settings in Macosx. Once you loaded those apps once it will be cached and changing the `Info.plist` does not affect the app. There was some way to force reload the cached settings but can't remember it and can't test.

Point is that all the KDE apps here that would be benefit from crispier fonts on Macos should get the `PlistBuddy` treatment as postprocessing OR we should propose upstream changes to the CMake files to do similar as Kate does (see https://github.com/KDE-mac/homebrew-kde/issues/315#issuecomment-578867957 ).
I guess it would be faster to fix things in here first. I'll see to open an MR for it soonish.

**Comment by yurikoles** [comment]
@bmihaila 

Thanks for investigation!

**Comment by yurikoles** [comment]
@bmihaila 

Any luck?

**Comment by bmihaila** [comment]
Hi @yurikoles 

Sorry, I did not have time to proceed with the Merge Request. Will try soon, but basically it is just adding those `PlistBuddy` lines mentioned in https://github.com/KDE-mac/homebrew-kde/issues/315#issuecomment-578790011 to each formula that uses fonts (guess that is more or less all?) and then testing them :). The last one is the more time consuming part. I have little time at the moment but will see what I can do to get things going.
